CMakeLists.txt

```diff
@@ -13,6 +13,8 @@ TRIBITS_PACKAGE_DECL(DataTransferKit)
 TRIBITS_ADD_DEBUG_OPTION()
 TRIBITS_ADD_SHOW_DEPRECATED_WARNINGS_OPTION()
 
+SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror")
+
```

Comment:
I think that neams wants to use `xlc` that's why I ask. If we decide that we don't support others compilers (I don't DTK has been built with any compilers but gcc, clang, and intel), then hard-coding the value is fine.